commit 899276a5aa9f07a2faf4e7b659c0ea5c90dbf3b9
Author: Nitzan Mordechai <nmordech@redhat.com>
Date:   Wed Jun 28 09:57:11 2023 +0000

    ceph-dencoder: osd - Add missing types
    
    Currently, ceph-dencoder lacks certain osd types, preventing us from accurately checking the ceph corpus for encode-decode mismatches.
    This pull request aims to address this issue by adding the missing types to ceph-dencoder.
    
    To successfully incorporate these types into ceph-dencoder, we need to introduce the necessary `dump` and `generate_test_instances`
    functions that was missing in some types. These functions are essential for proper encode and decode of the added types.
    
    This PR will enhance the functionality of ceph-dencoder by including the missing types, enabling a comprehensive analysis of encode-decode consistency.
    With the addition of these types, we can ensure the robustness and correctness of the ceph corpus.
    
    This update will significantly contribute to improving the overall reliability and accuracy of ceph-dencoder.
    It allows for a more comprehensive assessment of the encode-decode behavior, leading to enhanced data integrity and stability within the ceph ecosystem.
    
    Fixes: https://tracker.ceph.com/issues/61788
    Signed-off-by: Nitzan Mordechai <nmordech@redhat.com>

diff --git a/src/mgr/DaemonHealthMetric.h b/src/mgr/DaemonHealthMetric.h
index ce0dad2c87e..a6fbce29910 100644
--- a/src/mgr/DaemonHealthMetric.h
+++ b/src/mgr/DaemonHealthMetric.h
@@ -5,6 +5,7 @@
 
 #include <cstdint>
 #include <ostream>
+#include "common/Formatter.h"
 #include "include/denc.h"
 
 enum class daemon_metric : uint8_t {
@@ -66,7 +67,16 @@ public:
     denc(v.value.n, p);
     DENC_FINISH(p);
   }
-
+  void dump(Formatter *f) const {
+    f->dump_string("type", get_type_name());
+    f->dump_int("n", get_n());
+    f->dump_int("n1", get_n1());
+    f->dump_int("n2", get_n2());
+  }
+  static void generate_test_instances(std::list<DaemonHealthMetric*>& o) {
+    o.push_back(new DaemonHealthMetric(daemon_metric::SLOW_OPS, 1));
+    o.push_back(new DaemonHealthMetric(daemon_metric::PENDING_CREATING_PGS, 1, 2));
+  }
   std::string get_type_name() const {
     return daemon_metric_name(get_type());
   }
